### DISCUSSION TITLE
Testing in realm kotlin

### DISCUSSION THREAD

**Discussion by jhetrifork** [realm/realm-kotlin#948] [Q&A]
What are the plans for introducing a guide to testing projects using realm Kotlin? Can we expect any help from the official documentation any time soon?

**Comment by cmelchior** [comment]
We didn't have any concrete plans. Testing should ideally work out of the box with JVM support (with caveats around Multiplatform testing sometimes being awkward in general).

But if something is missing from our documentation we would be happy to add it. Are you looking for something particular?

**Reply by jhetrifork** [reply]
Hi @cmelchior I think something similar to what you have for the java sdk would be great: https://www.mongodb.com/docs/realm/sdk/java/advanced-guides/testing/

Some examples of how to code a basic unit test of code utilising Realm. If some testing frameworks are needed, showing a concrete example of how a unit or integration test could look like using those libraries with any necessary setup and teardown logic. I am currently trying to use mockk and robolectric to build some unit tests, and you would have a way better understanding of which caveats to think about when coding setup and teardown logic to be run between tests.

**Reply by cmelchior** [reply]
Yeah, It would probably make sense to have something like that. One of the reasons we had a fairly elaborate guide for Realm Java was that it didn't really run on Desktop, which is why you need to mock it in Roboletric.

With Realm Kotlin, we should support running on the JVM directly, which means that ideally, it should work out of the box. That said, we haven't done exploration around it (except running all our own unit tests), so we might have missed a few things that can be clarified.

I assume your project is Android only if you are using Robolectric?

**Reply by jhetrifork** [reply]
That makes a lot of sense. Yes I am doing Android only. :). As someone who is still learning how to test on Android I find it a bit hard to find my way through the many different testing library combinations, figuring out if ones a relevant or necessary and which ones obsolete. Even a small section with a basic example would be a massive help, especially if building test have become much simpler. 

Reading through old threads for the java sdk, it seems like the approach to testing realm code is something that has been widely discussed previously. I think having some clear guidance could benefit the adoption of the new sdk. Maybe going so far as outlining the differences between the java and kotlin sdk like you just did here could shed some light on the advantages with the new sdk. :)
